runtime/src/main/java/org/corfudb/runtime/LogReplicationLogicalGroupClient.java

```diff
@@ -34,6 +34,8 @@
 
 /**
  * A client to interface with log replication utilizing the logical groups replication model.
+ * <p>
+ * Thread safety of client operations enforced by utilization of transactions.
```

Comment:
I think this statement is not correct, the concurrency issue could be somehow concealed to some extend because we catch `TransactionAbortedException` and wrap a retry around it (like if 2 threads are modifying the same key concurrently). 

I think that's not deterministic and might be better to put method-level `synchronized` signature for methods that have side effects, if we do want to provide thread safety for this client

but open to ideas from others...